blockchain/validate.go

```diff
@@ -419,21 +419,21 @@ func CheckProofOfWork(block *bchutil.Block, powLimit *big.Int) error {
 // input and output scripts in the provided transaction.  This uses the
 // quicker, but imprecise, signature operation counting mechanism from
 // txscript.
-func CountSigOps(tx *bchutil.Tx) int {
+func CountSigOps(tx *bchutil.Tx, magneticAnomalyActive bool) int {
 	msgTx := tx.MsgTx()
 
 	// Accumulate the number of signature operations in all transaction
 	// inputs.
 	totalSigOps := 0
 	for _, txIn := range msgTx.TxIn {
```

Comment:
Had a chat with Josh during the hackathon.  I'm not sure if this is the right place, but sigops should not be counted for coinbase inputs.

Reply:
@jasonbcox it looks to me like abc is validating the coinbase input https://github.com/Bitcoin-ABC/bitcoin-abc/blob/efa9749bac3e20b2a8fc056c01d75e3dbc4e341d/src/validation.cpp#L1717

Reply:
I think the correct thing to do is only check the coinbase outputs. The input can be garbage. I think we hit an unlikely byte sequence and our validations triggered it. I want to try syncing with just coinbase input validation removed and see if that fixes the issue. Then we don't need to pass around booleans everywhere.

https://bitcoin.stackexchange.com/questions/4990/what-is-the-format-of-coinbase-input-scripts

Reply:
Actually I take that back. We need the mining interface to be accurate, and if ABC is counting those sigops even when the coinbase input has garbage data then we need to be consistent. I can investigate the `txScript.Engine` suggestion as I agree passing a boolean is a bit dirty.

That being said, my node is now stuck, so if we merge this change set then it unblocks people to start syncing, and we can refactor it to use the flags approach which will be a much better solution long term.